client.go

```diff
@@ -1360,20 +1360,27 @@ func validateQueueName(queueName string) error {
 //	if err != nil {
 //		// handle error
 //	}
-func (c *Client[TTx]) JobList(ctx context.Context, params *JobListParams) ([]*rivertype.JobRow, error) {
+func (c *Client[TTx]) JobList(ctx context.Context, params *JobListParams) ([]*rivertype.JobRow, *JobListCursor, error) {
```

Comment:
Agreed that the code needed to get the next cursor is a little overly verbose at the moment, and furthermore you have to make a `len` check on jobs to make sure you're not indexing a 0-length slice.

``` go
jobs, err := client.JobList(ctx, NewJobListParams().After(JobListCursorFromJob(jobs[len(jobs)-1])))
```

If we were going to change this API though, I think I'd be tempted to make it more like a result struct, which is a bit more conventional, and a little more futureproof. e.g.

``` go
type JobListResult struct {
    Jobs []*rivertype.JobRow
    NextCursor *JobListCursor
```

Thoughts?